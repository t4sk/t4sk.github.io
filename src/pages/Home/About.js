import React from "react"
import { Image, Icon, Divider } from "semantic-ui-react"
import styles from "./About.module.css"
import profile from "./profile.jpeg"

function About(props) {
  // TODO consulting and contact form

  return (
    <div className={styles.component}>
      <div className={styles.profile}>
        <Image
          src={profile}
          circular
          fluid
          style={{ width: 150, height: 150 }}
          alt="profile"
        />
        <h1 className={styles.header}>Tasuku Nakamura</h1>
      </div>
      <div className={styles.intro}>
        <p>
          Hi, I am a software engineer who makes videos about smart contracts.
        </p>
        <p>
          Creator of{" "}
          <a
            href="https://solidity-by-example.org/"
            target="__blank"
            rel="noopener noreferrer"
          >
            Solidity by Example
          </a>
        </p>
      </div>

      <Divider />

      <h3>Follow me</h3>
      <div>
        <a
          href="https://www.youtube.com/channel/UCJWh7F3AFyQ_x01VKzr9eyA/featured"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="youtube" color="red" size="large" />
            YouTube
          </div>
        </a>
        <a
          href="https://twitter.com/ProgrammerSmart"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="twitter" color="blue" size="large" />
            Twitter
          </div>
        </a>
        <a
          href="https://github.com/t4sk"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="github" color="black" size="large" />
            GitHub
          </div>
        </a>
      </div>

      <Divider />
      <p>
        You can contact me at{" "}
        <span className={styles.email}>
          contact@smartcontractprogrammer.com
        </span>
      </p>
    </div>
  )
}

About.propTypes = {}

export default About
