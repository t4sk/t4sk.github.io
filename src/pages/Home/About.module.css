.component {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-left: 20px;
  margin-right: 20px;
}
.header {
  font-size: 2.5rem;
}
.links {
  display: flex;
  flex-direction: row;
}
.intro {
  color: #4e4e4e;
  font-size: 1.3rem;
  margin-top: 50px;
}
.list {
  display: flex;
  flex-direction: column;
}
.listItem {
  margin-top: 5px;
  margin-bottom: 5px;
}
.email {
  font-weight: 700;
}
