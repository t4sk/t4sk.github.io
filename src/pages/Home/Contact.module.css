.email {
  font-weight: 700;
}
