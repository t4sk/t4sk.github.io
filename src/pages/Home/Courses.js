import React from "react"
import PropTypes from "prop-types"
import styles from "./Courses.module.css"
import CourseCard from "./CourseCard"

import learnSolidityCardLogo from "../../static/learn-solidity-card.svg"

const COURSES = [
  {
    title: "Learn Solidity",
    logo: learnSolidityCardLogo,
    href: "/courses/learn-solidity",
  },
]

export function Courses(props) {
  return (
    <div className={styles.component}>
      {COURSES.map(course => (
        <CourseCard key={course.title} {...course} />
      ))}
    </div>
  )
}

Courses.propTypes = {}

export default Courses
