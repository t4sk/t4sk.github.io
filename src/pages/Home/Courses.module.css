.component {
  background: #f4f7f9;
}
