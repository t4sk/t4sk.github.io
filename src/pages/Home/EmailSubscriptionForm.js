import React from "react"
import PropTypes from "prop-types"
import { Input } from "semantic-ui-react"
import styles from "./EmailSubscriptionForm.module.css"

export function EmailSubscriptionForm(props) {
  return (
    <div className={styles.component}>
      <h1>Notify me of new tutorials</h1>
      <div className={styles.input}>
        <Input
          action={{
            // color: "green",
            content: <div className={styles.action}>Subscribe</div>,
          }}
          fluid
          size="huge"
          placeholder="Type your email..."
        />
      </div>
    </div>
  )
}

EmailSubscriptionForm.propTypes = {}

export default EmailSubscriptionForm
