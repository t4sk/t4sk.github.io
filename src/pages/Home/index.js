import React from "react"
import PropTypes from "prop-types"
import Layout from "../../components/Layout"
import SEO from "../../components/SEO"

// TODO list most recent posts (both eng and jp)
function Home(props) {
  const { lang } = props

  return (
    <Layout lang={lang}>
      <SEO title="Smart Contract Programmer" lang={lang} />
      <div>{lang}</div>
    </Layout>
  )
}

Home.propTypes = {
  lang: PropTypes.oneOf(["en", "jp"]).isRequired,
}

export default Home
