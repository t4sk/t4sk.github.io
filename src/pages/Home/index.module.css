.main {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.posts {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}
.row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.post {
  margin: 10px;
  margin-top: 35px;
  flex-grow: 1;
}
