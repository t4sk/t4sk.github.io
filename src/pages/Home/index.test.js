import React from "react"
import { shallow } from "enzyme"
import Home from "./index"

const props = {
  lang: "en",
}

test("en", () => {
  const component = shallow(<Home.En {...props} />)

  expect(component).toMatchSnapshot()
})

test("jp", () => {
  const component = shallow(<Home.Jp {...props} lang="jp" />)

  expect(component).toMatchSnapshot()
})
