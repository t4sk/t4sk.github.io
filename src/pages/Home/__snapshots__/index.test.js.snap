// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<PageLayout>
  <SEO
    description="Learn how to program smart contracts."
    keywords={
      Array [
        "smart contract",
        "solidity",
        "ethereum",
      ]
    }
    lang="en"
    meta={Array []}
    title="Smart Contract Programmer"
  />
  <Segment
    inverted={true}
    style={
      Object {
        "backgroundColor": "#20232a",
        "minHeight": 150,
        "padding": "2em 0em",
      }
    }
    textAlign="center"
    vertical={true}
  >
    <Container
      text={true}
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Image
          alt="logo"
          as="img"
          src="logo.svg"
          style={
            Object {
              "height": 250,
            }
          }
          ui={true}
        />
      </div>
      <Header
        as="h1"
        content="Smart Contract Programming"
        inverted={true}
        style={
          Object {
            "fontSize": "2.5em",
            "marginBottom": 0,
            "marginTop": 0,
          }
        }
      />
      <p>
        a blog for developers about smart contracts
      </p>
    </Container>
  </Segment>
  <div
    className="main"
  >
    <div
      className="posts"
    >
      <div
        className="row"
        key="0"
      >
        <div
          className="post"
          key="0"
        >
          <Post
            createdAt="2019-12-07"
            thumbnailUrl="https://i.ytimg.com/vi/hzxdGvcsDvU/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCOTTmKP0noS7RPRHiBz9trPRFwEw"
            title="Learn Solidity - Loop"
            youTubeUrl="https://www.youtube.com/watch?v=hzxdGvcsDvU"
          />
        </div>
        <div
          className="post"
          key="1"
        >
          <Post
            createdAt="2019-12-01"
            thumbnailUrl="https://i.ytimg.com/vi/k4mdv97up5M/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCCuy97YgJvXzihSTt4jwpAHM-dFQ"
            title="Learn Solidity - Error"
            youTubeUrl="https://www.youtube.com/watch?v=k4mdv97up5M"
          />
        </div>
        <div
          className="post"
          key="2"
        >
          <Post
            createdAt="2019-11-25"
            thumbnailUrl="https://i.ytimg.com/vi/BduMOagAuKs/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLBQV4qUgy6AHOR5GZYnuJxt-tM6aQ"
            title="Learn Solidity - Events"
            youTubeUrl="https://www.youtube.com/watch?v=BduMOagAuKs"
          />
        </div>
      </div>
      <div
        className="row"
        key="1"
      >
        <div
          className="post"
          key="0"
        >
          <Post
            createdAt="2019-11-15"
            thumbnailUrl="https://i.ytimg.com/vi/NBzQVJ6OrrQ/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLAHT2JRAbMuSeLilA2TVeepyfuzcg"
            title="Learn Solidity - Visibility"
            youTubeUrl="https://www.youtube.com/watch?v=NBzQVJ6OrrQ&t=2s"
          />
        </div>
        <div
          className="post"
          key="1"
        >
          <Post
            createdAt="2019-11-08"
            thumbnailUrl="https://i.ytimg.com/vi/dcNpIOhTNQg/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCjHeYDhI4TvPhI8TnnKFJN4sJOpA"
            title="Learn Solidity - Inheritance (part 3)"
            youTubeUrl="https://www.youtube.com/watch?v=dcNpIOhTNQg"
          />
        </div>
        <div
          className="post"
          key="2"
        >
          <Post
            createdAt="2019-11-04"
            thumbnailUrl="https://i.ytimg.com/vi/Q5_Gmm_IZSI/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCjGnK1kE6_65BGCnXFzYL10ES4Bg"
            title="Learn Solidity - Inheritance (part 2)"
            youTubeUrl="https://www.youtube.com/watch?v=Q5_Gmm_IZSI"
          />
        </div>
      </div>
      <div
        className="row"
        key="2"
      >
        <div
          className="post"
          key="0"
        >
          <Post
            createdAt="2019-10-30"
            thumbnailUrl="https://i.ytimg.com/vi/GnztHR8-1o8/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLBQeYOr7TI-FFZDyz06dhADsv7JXQ"
            title="Learn Solidity - Inheritance (part 1)"
            youTubeUrl="https://www.youtube.com/watch?v=GnztHR8-1o8"
          />
        </div>
        <div
          className="post"
          key="1"
        >
          <Post
            createdAt="2019-10-24"
            thumbnailUrl="https://i.ytimg.com/vi/HpjZZ9tASGs/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLBmmxQO823nu0rRRnQNg9IAglG4cw"
            title="Learn Solidity - Constructor"
            youTubeUrl="https://www.youtube.com/watch?v=HpjZZ9tASGs"
          />
        </div>
        <div
          className="post"
          key="2"
        >
          <Post
            createdAt="2019-10-20"
            thumbnailUrl="https://i.ytimg.com/vi/thADMg9cKPM/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDZkbd-AkQwPSSQ7rIejxHSJhzU7g"
            title="Learn Solidity - Function Modifiers"
            youTubeUrl="https://www.youtube.com/watch?v=thADMg9cKPM"
          />
        </div>
      </div>
      <div
        className="row"
        key="3"
      >
        <div
          className="post"
          key="0"
        >
          <Post
            createdAt="2019-10-12"
            thumbnailUrl="https://i.ytimg.com/vi/xknoxALAL8c/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLChZTh-2zMGnYu_pF0AD5XREgaB8w"
            title="Learn Solidity - View and Pure Functions"
            youTubeUrl="https://www.youtube.com/watch?v=xknoxALAL8c"
          />
        </div>
        <div
          className="post"
          key="1"
        >
          <Post
            createdAt="2019-10-06"
            thumbnailUrl="https://i.ytimg.com/vi/71cmPaD_AnQ/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCcIoVXOmZYekcdlbijNKfCD7KLoA"
            title="Learn Solidity - Functions"
            youTubeUrl="https://www.youtube.com/watch?v=71cmPaD_AnQ"
          />
        </div>
        <div
          className="post"
          key="2"
        >
          <Post
            createdAt="2019-09-30"
            thumbnailUrl="https://i.ytimg.com/vi/oTS9uxU6cAM/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLBFNU6ibn-6fR2RUzTKjSSowdAAoA"
            title="Learn Solidity - Gas and Gas Price"
            youTubeUrl="https://www.youtube.com/watch?v=oTS9uxU6cAM"
          />
        </div>
      </div>
      <div
        className="row"
        key="4"
      >
        <div
          className="post"
          key="0"
        >
          <Post
            createdAt="2019-09-21"
            thumbnailUrl="https://i.ytimg.com/vi/ybPQsjssyNw/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLBS8cUaXPkG9N9EJWSwHCchOI-NNQ"
            title="Learn Solidity - Ether and Wei"
            youTubeUrl="https://www.youtube.com/watch?v=ybPQsjssyNw"
          />
        </div>
        <div
          className="post"
          key="1"
        >
          <Post
            createdAt="2019-09-15"
            thumbnailUrl="https://i.ytimg.com/vi/4XQsHBJScEk/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLCSidkVzqzY9matqnVSmYJZNO8IAA"
            title="Learn Solidity - State Variables"
            youTubeUrl="https://www.youtube.com/watch?v=4XQsHBJScEk&t=5s"
          />
        </div>
        <div
          className="post"
          key="2"
        >
          <Post
            createdAt="2019-08-20"
            thumbnailUrl="https://i.ytimg.com/vi/hMwdd664_iw/hqdefault.jpg?sqp=-oaymwEZCPYBEIoBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDZXbU3wGs1vpsnWX06BEHw9TookQ"
            title="Learn Solidity - Hello World"
            youTubeUrl="https://www.youtube.com/watch?v=hMwdd664_iw"
          />
        </div>
      </div>
    </div>
  </div>
</PageLayout>
`;
